src/sentry/web/frontend/vercel_extension_configuration.py

```diff
@@ -1,6 +1,25 @@
+from urllib.parse import parse_qsl, urlparse
+
+from django.http import HttpResponseRedirect
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.models import Integration
+from sentry.utils.http import absolute_uri
+
 from .integration_extension_configuration import IntegrationExtensionConfigurationView
 
 
 class VercelExtensionConfigurationView(IntegrationExtensionConfigurationView):
     provider = "vercel"
     external_provider_key = "vercel"
+
+    def redirect(self, request: Request) -> Response:
+        configuration_id = dict(parse_qsl(urlparse(request).query))["configurationId"]
+        integration = Integration.objects.get(
+            provider="vercel", metadata__contains=configuration_id
```

Comment:
This query felt kind of shaky to me, but the only information we have is the configuration ID (which we store in the metadata as the installation ID) and the user from the request. I could validate on the user as well by checking that they are in the organization, but I wasn't sure if that was necessary.